MekHQ/src/mekhq/campaign/finances/Loan.java

```diff
@@ -310,17 +313,35 @@ public static int[] getCollateralBracket(final int rating) {
      * Determines the maximum number of years by clamping the given rating to a valid range.
      *
      * <p>This method returns a value that ensures the input {@code rating} falls within the specified
-     * range of 1 to 7. Ratings below 1 are clamped to 1, and ratings above 7 are clamped to 7. The
-     * clamped value is directly returned.</p>
+     * range from the loan table. Ratings below 1 are clamped to 1, and ratings above 7 are clamped to 7. The clamped
+     * value is directly returned. The '6 months' period is currently rounded up to 1 year. See CO 4th p50, Loan
+     * Table</p>
+     * <p>TODO: Change slider and results to months to allow for 6 month periods</p>
+     * <br>
      *
-     * <p>The clamped values coincide with the Experience Level ordinals (Ultra-Green, Green, etc).
-     * This means a Veteran-rated campaign (ordinal 4) could take up to a 4-year loan.</p>
+     * @param rating   the input rating value to be clamped.
+     * @param campaign the campaign object for faction type
      *
-     * @param rating the input rating value to be clamped.
      * @return the clamped rating, guaranteed to be a value between 1 and 7 (inclusive).
      */
-    public static int getMaxYears(int rating) {
-        return clamp(rating, 1, 7);
+    public static int getMaxYears(int rating, Campaign campaign) {
+        int paybackPeriod = 1;
+        if (campaign.getFaction().isGovernment()) {
+            paybackPeriod = 10;
+        } else if (campaign.getFaction().isPirate()) {
+            paybackPeriod = 1;
+        } else if (rating <= 0) {
+            paybackPeriod = 1;
+        } else if (rating < 5) {
+            paybackPeriod = 1;
+        } else if (rating < 10) {
+            paybackPeriod = 2;
+        } else if (rating < 14) {
+            paybackPeriod = 3;
+        } else {
+            paybackPeriod = 5;
+        }
+        return clamp(rating, 1, paybackPeriod);
```

Comment:
This doesnt make much sense, if the faction matters then you pass the faction as argument, not the campaign.
You are then also doing some weird calculations were rating is a number between 0 and 14, I dont know what you are expecting the rating to be, but it really doesnt look like it is correct.

There is also no TODO in here, so I dont know what motivated you to change it.